pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1537,6 +1537,20 @@ type ObjectStoreSpec struct {
 	// +nullable
 	// +optional
 	Hosting *ObjectStoreHostingSpec `json:"hosting,omitempty"`
+
+	// RGWReadAffinity defines the read affinity policy to optimize the read requests for the RGW clients
+	RGWReadAffinity *RGWReadAffinity `json:"rgwReadAffinity,omitempty"`
+}
+
+// RGWReadAffinity defines the read affinity policy to optimize the read requests for the RGW clients
+type RGWReadAffinity struct {
+	// localize: read from the nearest OSD based on crush location of the RGW client
+	// balance: picks a random OSD from the PG's active set
+	// default: read from the primary OSD
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=localize;balance;default
```

Comment:
Enums can help users, but they can also make it harder to keep up with Ceph changes, when Ceph introduces new options for this. Do we anticipate Ceph adding new options?